### PR TITLE
Release v2.10.0

### PR DESCRIPTION
**Pull request by zhiqiangjin**
### Notes:
- Supported Controller version: **UserConnect-5.2.2122** 
- Supported Terraform version: **v0.12.x**

### Features:
1. Implemented advanced VPC attachment options for both **aviatrix_aws_tgw** and **aviatrix_aws_tgw_vpc_attachment**
2. Implemented support for updating ``customized_routes`` in **aviatrix_aws_tgw_vpc_attachment**
3. Implemented string length verification for ``aws_account_number`` in **aviatrix_account**
4. Implemented support for ``customized_spoke_vpc_routes``, ``filtered_spoke_vpc_routes`` and ``include/exclude_advertised_spoke_routes`` options in **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**
5. Implemented support for configuring CloudN backup for controller in **aviatrix_controller_config**
6. New resources:
  - **aviatrix_gateway_dnat**
  - **aviatrix_gateway_snat**
7. New data sources:
  - **aviatrix_spoke_gateway**
  - **aviatrix_transit_gateway**
  - **aviatrix_firenet**

### Enhancements:
1. Added coverage for **aviatrix_gateway_dnat** and **aviatrix_gateway_snat** in test-infra 
2. Added coverage for **data_source_aviatrix_spoke_gateway**, **data_source_aviatrix_transit_gateway** and **data_source_aviatrix_firenet** in test-infra 
3. Deprecated ``dnat_policy`` in **aviatrix_gateway**
4. Deprecated ``dnat_policy``, ``snat_policy`` and ``snat_mode`` in **aviatrix_spoke_gateway** 
5. Replaced ``enable_snat`` with ``single_ip_snat`` in **aviatrix_gateway**, **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**

### Bug Fixes:
1. Fixed issue where importing the **aviatrix_aws_tgw** resource results in deltas that could not be rectified through apply

### CODE REVIEW

**Review by arvindsree — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,32 @@
...
+### Notes:
+- Supported Controller version: **UserConnect-5.2.2122** 
+- Supported Terraform version: **v0.12.x**
+
+### Features:
+1. Implemented advanced VPC attachment feature in both **aviatrix_aws_tgw** and **aviatrix_aws_tgw_vpc_attachment**
+2. Implemented updating ``customized_routes`` in **aviatrix_aws_tgw_vpc_attachment**
+3. Implemented a check that ``aws_account_number`` needs to be exactly 12 digits in **aviatrix_account**
+4. Implemented multiple missing parameters in **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**
+5. Implemented functionality of enabling/disabling cloudn backup configuration for controller in **aviatrix_controller_config**
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: 5. Implemented support for configuring CloudN backup for controller in **aviatrix_controller_config***

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,4 +1,32 @@
-## 2.9.2 (Unreleased)
+## 2.10.0 (Unreleased)
+### Notes:
+- Supported Controller version: **UserConnect-5.2.2122** 
+- Supported Terraform version: **v0.12.x**
+
+### Features:
+1. Implemented advanced VPC attachment feature in both **aviatrix_aws_tgw** and **aviatrix_aws_tgw_vpc_attachment**
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: 1. Implemented advanced VPC attachment options for both **aviatrix_aws_tgw** and **aviatrix_aws_tgw_vpc_attachment**

**Comment by arvindsree** [comment]
typo in title

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,32 @@
...
+2. Implemented updating ``customized_routes`` in **aviatrix_aws_tgw_vpc_attachment**
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: 2. Implemented support for updating ``customized_routes`` in **aviatrix_aws_tgw_vpc_attachment**

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,4 +1,32 @@
...
+3. Implemented a check that ``aws_account_number`` needs to be exactly 12 digits in **aviatrix_account**
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: 3. Implemented string length verification for ``aws_account_number`` in **aviatrix_account**

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,4 +1,32 @@
...
+4. Implemented multiple missing parameters in **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: Maybe specify which attributes? or categorize it to maybe "route filtering and advertising options"

For example:
4. Implemented support for spoke VPC route filtering and advertising options in **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,4 +1,32 @@
...
+
+### Enhancements:
+1. Added coverage for **aviatrix_gateway_dnat** and **aviatrix_gateway_snat** in test-infra 
+2. Added coverage for **data_source_aviatrix_spoke_gateway**, **data_source_aviatrix_transit_gateway** and **data_source_aviatrix_firenet** in test-infra 
+3. Deprecated ``dnat_policy`` in **aviatrix_gateway**
+4. Deprecated ``dnat_policy``, ``snat_policy`` and ``snat_mode`` in **aviatrix_spoke_gateway** 
+5. Replaced ``enable_snat`` with ``single_ip_snat`` in **aviatrix_gateway**, **aviatrix_spoke_gateway** and **aviatrix_transit_gateway**
+
+### Bug Fixes:
+1. Fixed issue where exporting TGW and importing leave deltas not rectified through apply in **aviatrix_aws_tgw**
```
Comment (inc1t3Ful) [inline CHANGELOG.md]: 1. Fixed issue where importing the **aviatrix_aws_tgw** resource results in deltas that could not be rectified through apply

**Pull request merged**